packages/devtools_app/lib/src/config_specific/launch_url/_launch_url_web.dart

```diff
@@ -0,0 +1,11 @@
+// Copyright 2021 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be found
+// in the LICENSE file.
+
+import 'dart:html';
+
+final isVSCodeEmbedded = window.navigator.userAgent.contains('Electron');
```

Comment:
It might be nice to have this somewhere more shared and use it here too:

https://github.com/flutter/devtools/blob/4e1ccbec3f95f0cd30fb51754845d95c9034c68b/packages/devtools_app/lib/src/config_specific/framework_initialize/_framework_initialize_web.dart#L60

Reply:
Would it better to inject a param from DartCode, to make sure it's running in VSCode? I guess it's possible there would be some other Electron-based browser running DevTools

Reply:
In the other code (where keystrokes are passed up) this check was to prevent another app hosting DevTools and enabling this (see https://github.com/flutter/devtools/issues/2775#issuecomment-794266068). It's probably less of a concern here, though it might be good to be consistent (I'll defer to @jacob314!).

Reply:
There is also the existing query parameter passed in that indicates whether to use the embedded version of DevTools. That could be reasonable to use in combination with checking for electron to ensure that we only show in VSCode.